clang/test/CIR/Lowering/ThroughMLIR/for.cir

```diff
@@ -0,0 +1,222 @@
+// RUN: cir-opt %s -cir-to-mlir --canonicalize | FileCheck %s -check-prefix=MLIR
+// RUN: cir-opt %s -cir-to-mlir --canonicalize -cir-mlir-to-llvm | mlir-translate -mlir-to-llvmir | FileCheck %s -check-prefix=LLVM
+
+!s32i = !cir.int<s, 32>
+module {
+  cir.global external @a = #cir.zero : !cir.array<!s32i x 100>
+
```

Comment:
Can you add a comment with the original C/C++ code you are building or write the test in a way that is testable all the way from the source file? As we change CIR, I'm afraid these tests are going to get hard to update. We already have that problem elsewhere, but probably best not to increase them. Similar for your other PR. No need to tackle right now, a future PR sounds good.

Reply:
Thanks for letting me know. I will create a PR to update for.cir soon.